src/bundle/Resources/public/js/scripts/admin.location.add.customurl.js

```diff
@@ -0,0 +1,29 @@
+(function (global, doc) {
+    const modal = doc.querySelector('#ez-modal--custom-url-alias');
+    const discardBtns = [...modal.querySelectorAll('[data-dismiss="modal"]')];
+    const submitBtn = modal.querySelector('[type="submit"]');
+    const inputs = [...modal.querySelectorAll('[required="required"]')];
+    const checkboxes = [...modal.querySelectorAll('.ez-field-edit--ezboolean input')];
+    const toggleButtonState = () => inputs.forEach(input => {
+        const hasValue = input.value.trim().length !== 0;
+        const methodName = hasValue ? 'removeAttribute' : 'setAttribute';
+
+        submitBtn[methodName]('disabled', true);
+    });
+    const toggleCheckbox = (event) => {
+        const checkbox = event.target;
+        const methodName = checkbox.checked ? 'add' : 'remove';
+        const checkedMethodName = checkbox.checked ? 'setAttribute' : 'removeAttribute';
+
+        checkbox.closest('.ez-data-source__label').classList[methodName]('is-checked');
+        checkbox[checkedMethodName]('checked', true);
```

Comment:
What is the point of setting the `checked` attr if the event is `change`?

Reply:
That was surprising for me as well, but without setting it explicite it was just not working. Going to check that again after @sunpietro improvements.